### PR TITLE
chore(ci): is build required?

### PR DESCRIPTION
**Pull request by mcdurdin**
Adds a script that tests whether changes are likely to impact the target platform and cancels the build if not.

This has been integrated as a build step in all of the Test build configurations.

### CODE REVIEW

File: resources/build/build-triggers.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+#
+# This file maps specific paths to build triggers
+#
+
+# the base folder for each pattern does not need to be included, nor oem folders
+# e.g. android='common/predictive-text|common/lexical-model-types'
+# will expand into android='^(android|(oem/[^/]+/android)|common/predictive-text|common/lexical-model-types)'
+
+watch_android='android|common/predictive-text|common/lexical-model-types'
+watch_ios='common/predictive-text|common/lexical-model-types'
```
Comment (darcywong00) [inline resources/build/build-triggers.sh]: Don't android and ios also need to watch for `web`?

File: resources/build/build-triggers.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+watch_linux='common/engine'
+watch_mac='common/engine'
```
Comment (darcywong00) [inline resources/build/build-triggers.sh]: Is this for future feature when mac uses Keyman core?

**Comment by darcywong00** [comment]
32f220cf2224b69a748dcd71dd447905b431972f worked for `is-build-required.sh`, but I'm not seeing executable bit on `build-triggers.sh` or `trigger-builds.sh`

**Comment by mcdurdin** [comment]
> but I'm not seeing executable bit on `build-triggers.sh` or `trigger-builds.sh`

They shouldn't need executable bits because they are source-included (e.g. `. ./build-triggers.sh`)

File: resources/build/build-triggers.sh (hunk 2 above)
Comment (mcdurdin) [reply resources/build/build-triggers.sh]: Yes, also to force builds whenever Keyman core is built (this reminds me that I want to rename that folder to common/core!)

File: resources/build/build-triggers.sh (hunk 1 above)
Comment (mcdurdin) [reply resources/build/build-triggers.sh]: Done.

**Pull request merged**